packages/clay-time-picker/src/index.tsx

```diff
@@ -4,57 +4,422 @@
+		return newValue;
+	};
+
+	const handleKeyDown = (
+		event: React.KeyboardEvent<HTMLInputElement>,
+		value: string,
+		configName: TimeType
+	) => {
+		const config = useConfig[configName];
+		const intrinsicValue = Number(value);
+		const setValue = (newValue: string) =>
+			onInputChange({
+				...values,
+				// eslint-disable-next-line sort-keys
+				[configName]:
+					configName === TimeType.ampm
+						? newValue
+						: handleMaxAndMin(newValue, config as ConfigMaxMin),
+			});
+
+		switch (event.keyCode) {
+			case KEY_BACKSPACE:
+				setValue(DEFAULT_VALUE);
+				break;
+			case KEY_ARROWRIGHT:
+				if (minutesRef.current) {
+					minutesRef.current.focus();
+				}
+				break;
+			case KEY_ARROWLEFT:
+				if (hoursRef.current) {
+					hoursRef.current.focus();
+				}
+				break;
+			case KEY_ARROWUP:
+				if (configName !== TimeType.ampm) {
+					if (value !== DEFAULT_VALUE) {
+						setValue(String(intrinsicValue + 1));
+					} else {
+						setValue(String((config as ConfigMaxMin).min));
+					}
+				} else {
+					setValue((config as ConfigAmpm).pm);
+				}
+				break;
+			case KEY_ARROWDOWN:
+				if (configName !== TimeType.ampm) {
+					if (value !== DEFAULT_VALUE && intrinsicValue !== 0) {
+						setValue(String(intrinsicValue - 1));
+					} else {
+						setValue(String((config as ConfigMaxMin).max));
+					}
+				} else {
+					setValue((config as ConfigAmpm).am);
+				}
+				break;
+			default:
+				if (regex.test(event.key) && configName !== TimeType.ampm) {
+					if (value === DEFAULT_VALUE) {
+						setValue(event.key);
+					} else {
+						setValue((value ? value : '') + event.key);
+					}
+				}
+				break;
+		}
+	};
+
+	const handleAction = (direction: number) => {
+		const {configName} = elementFocusedRef.current;
+		const config = useConfig[configName];
+		const prevValue = values[configName];
+		const value =
+			configName === TimeType.ampm
+				? direction === 1
+					? (config as ConfigAmpm).pm
+					: (config as ConfigAmpm).am
+				: handleMaxAndMin(
+						String(
+							(prevValue !== DEFAULT_VALUE
+								? Number(prevValue)
+								: 0) + direction
+						),
+						config as ConfigMaxMin
+				  );
 
-	return timePickerContent(otherProps);
+		onInputChange({
+			...values,
+			// eslint-disable-next-line sort-keys
+			[configName]: value,
+		});
+	};
+
+	const handleDocumentClick = (event: Event) => {
+		if (
+			elementRef.current &&
+			event.target !== null &&
+			!elementRef.current.contains(event.target as HTMLDivElement)
+		) {
+			elementFocusedRef.current = defaultElementFocus;
+			setActionVisible(false);
+			setIsFocused(false);
+		}
+	};
+
+	useEffect(() => {
+		document.addEventListener('click', handleDocumentClick);
+
+		return () => {
+			document.removeEventListener('click', handleDocumentClick);
+		};
+	}, []);
+
+	const visibleActionReset =
+		actionVisible &&
+		(values.ampm !== DEFAULT_VALUE ||
+			values.hours !== DEFAULT_VALUE ||
+			values.minutes !== DEFAULT_VALUE);
+
+	return (
```

Comment:
Theres quite a bit of logic and function creation above the actual return, we might want to further abstract to help readability of the logic flow.